R/writeComps.R

```diff
@@ -456,20 +460,45 @@ writeComps = function(inComps, fname = NULL, abins = NULL, lbins = NULL,
   # IGT(2019-04-25): I tried using an apply function but kept messing up,
   # so fell back on a simple loop over the rows
   rescale.comps <- function(out){
-    value.names <- grep("^[alfmALFM][0-9]+", colnames(out), value = TRUE)
+    value.names <- grep("^[alfmuALFMU][0-9]+", colnames(out), value = TRUE)
     for(irow in 1:nrow(out)){
       out[irow, names(out) %in% value.names] <-
         out[irow, names(out) %in% value.names] /
           sum(out[irow, names(out) %in% value.names])
     }
+    # Code to apply the rescaled comps to the second matrix
+    # for Mout, Fout, or Uout composition data.
+    if ("ageErr" %in% colnames(out)){
+      adj <- 11
+    } else {
+      adj <- 8
+    }
+    if (length(value.names) < (dim(out)[2] - adj)){
```

Comment:
add comment to say what these lines (476-480) are doing?

Reply:
Submitted before writing summary note: all changes look good to me. I haven't run the revised code so am trusting that it has been tested.
@kellijohnson-NOAA should be added as an author on the `writeComps()` function.

Reply:
Submitted before writing summary note: all changes look good to me. I haven't run the revised code so am trusting that it has been tested.
[@kellijohnson-NOAA](https://github.com/kellijohnson-NOAA) should be added as an author on the writeComps() function.

Reply:
I have tested the function for both length and age composition data creation for Dover sole.  I will add @kellijohnson-NOAA to the author list here.

Reply:
Description of what is occurring in the if statement and updated the author list in commit 8693650cd6e3dbdeac3045f1ca7cea478f2cab8f